### DISCUSSION TITLE
Bug: No Date/Time picker for Libre sensor when "Not today" is selected and start time is reported 24 hours in the past

### DISCUSSION THREAD

**Discussion by jees1234** [NightscoutFoundation/xDrip#3560] [General]
Hello,

I have noticed that when I start new Libre sensor and select "Not today" there is no Date/Time picker to select exact date and time when sensor was started. Also the sensor start time in logs is reported 1 day in the past.
E.g. I have changed sensor today 2019-08-12 but in xDrip+ logs there is:

NEW SENSOR
14:24 Sensor started at 2019-08-11 14:24:35

This also affects Nightscout upload as it is also uploaded incorrectly - see entry from MongoDB:

{ "_id" : ObjectId("0e84a2f5dc2d405ebb7a3cbc"),
"timestamp" : 1565529875527,
"eventType" : "Sensor Start",
"enteredBy" : "xdrip",
"uuid" : "",
"created_at" : "2019-08-11T13:24:35.000Z",  <-----
"sysTime" : "2019-08-11T15:24:35.527+0200",  <-----
"utcOffset" : 0 }

Hence it affects SAGE plugin as sensor age is reported incorrectly.

![nightscout_sage](https://user-images.githubusercontent.com/51976401/62871339-9ef2f500-bd1b-11e9-8d85-441d1ee954f7.png)

I have briefly checked xDrip+ source code and found possible source of the issue. In StartNewSensor.java there is:

if (DexCollectionType.hasLibre()) {
                    ucalendar.add(Calendar.DAY_OF_MONTH, -1);
                    realStartSensor();
                } else { 

Not sure why it is there and date is shifted 1 day back.
I think there should be code to display Date/Time picker as user can select exact date and time when sensor was started.

Also I'm a little bit confused why sensor start time is adjusted by -1 hour when option "Yes, today" is selected and Time has been picked up in dialog:

ucalendar.add(Calendar.HOUR_OF_DAY, -1); // hack for warmup time 

It is questionable whether is it OK because the dialog says "What time was it inserted?" So I would expect that I should enter time when sensor was inserted and started by Libre reader/LibreLink app. If so the code above actually makes sensor start time 1 hour earlier than it was actually.
In my opinion this is confusing and it should be removed.
Thanks.

**Comment by Navid200** [comment]
@psonnera  Would you please provide your view of this?

**Comment by psonnera** [comment]
@Navid200 I nearly never restart the sensor (logically I mean) and also never checked in Nightscout as the actual SAGE I was convinced was picked from the real sensor lifetime...
The rest of the discussion is indeed vocabulary accuracy and historical reasons. You can insert a Libre sensor and start it, at this point the logic is sane. But you can insert a sensor and not start it immediately. So I'd agree that the question should be "What time was it started?" since starting a Libre means you need to wait for an hour before you can get consistent readings (and you can get inconsistent readings during warmup if you like crazy numbers).
I would remove the dialog and rely on sensor information from FRAM. Start sensor should only be a way to say there's a new sensor, no time necessary as long as the information can be grabbed from the sensor FRAM (but don't rely on me to remember if this is always feasible, but I'd guess yes).
